airbyte-workers/src/main/java/io/airbyte/workers/temporal/scheduling/activities/RecordMetricActivityImpl.java

```diff
@@ -86,9 +99,23 @@ private Map<String, Object> generateTags(final ConnectionUpdaterInput connection
       if (connectionUpdaterInput.getJobId() != null) {
         tags.put(JOB_ID_KEY, connectionUpdaterInput.getJobId());
       }
+      workspaceIdOptional.ifPresent(id -> tags.put(WORKSPACE_ID_KEY, id));
     }
 
     return tags;
   }
 
+  private Optional<UUID> getWorkspaceId(final RecordMetricInput recordMetricInput) {
+    if (recordMetricInput.getConnectionUpdaterInput() == null) {
+      return Optional.empty();
+    }
+    try {
+      final UUID connectionId = recordMetricInput.getConnectionUpdaterInput().getConnectionId();
+      final WorkspaceRead workspaceRead = workspaceApi.getWorkspaceByConnectionId(new ConnectionIdRequestBody().connectionId(connectionId));
```

Comment:
I believe that it would worth implementing some caching here. The workspace is not expecting to change during the run so we will do many useless request. I think that a in memory cache can be possible. Another solution would be to add an activity that will fetch the workspace Id from a connection Id. We currently have ~8 calls to the record metrics which will generate 8 calls where we actually need only one.

Reply:
Added @Cacheable in order to use a simple Micronaut cache for this method

Reply:
I'm not very familiar with the Cacheable annotation here. I would defer to Benoit for this.